api/taskgroup.go

```diff
@@ -208,6 +207,16 @@ func (h TaskGroupHandler) Update(ctx *gin.Context) {
 			return
 		}
 	case tasking.Ready:
+		for i := range m.Tasks {
+			task := &m.Tasks[i]
+			if task.ID > 0 {
+				_ = ctx.Error(
+					&BadRequestError{
+						Reason: "already submitted.",
+					})
+				return
+			}
+		}
```

Comment:
hint: Since the update of a _Ready_ task effectively _submits_ it which creates all of the tasks, this check ensure the tasks are not created more than once.